operator/pkg/ingress/ingress_class.go

```diff
@@ -4,172 +4,181 @@
 package ingress
 
 import (
+	"context"
 	"fmt"
+	"strconv"
+	"sync/atomic"
 
-	"k8s.io/apimachinery/pkg/util/wait"
-	"k8s.io/client-go/tools/cache"
 	"k8s.io/client-go/util/workqueue"
 
-	"github.com/cilium/cilium/pkg/k8s"
-	k8sClient "github.com/cilium/cilium/pkg/k8s/client"
-	"github.com/cilium/cilium/pkg/k8s/informer"
+	"github.com/cilium/cilium/pkg/k8s/resource"
 	slim_networkingv1 "github.com/cilium/cilium/pkg/k8s/slim/k8s/api/networking/v1"
-	"github.com/cilium/cilium/pkg/k8s/utils"
 	"github.com/cilium/cilium/pkg/logging/logfields"
 )
 
-type ingressClassAddedEvent struct {
-	ingressClass *slim_networkingv1.IngressClass
-}
-
-type ingressClassUpdatedEvent struct {
-	oldIngressClass *slim_networkingv1.IngressClass
-	newIngressClass *slim_networkingv1.IngressClass
-}
-
-type ingressClassDeletedEvent struct {
-	ingressClass *slim_networkingv1.IngressClass
-}
-
-type ingressClassManager struct {
-	informer cache.Controller
-	store    cache.Store
-
-	queue      workqueue.RateLimitingInterface
-	maxRetries int
-
-	ingressQueue workqueue.RateLimitingInterface
-}
-
-// type used to signal changes to the ingress controller queue
 type ciliumIngressClassUpdatedEvent struct {
-	ingressClass *slim_networkingv1.IngressClass
+	isDefault bool
+	changed   bool
 }
 
 type ciliumIngressClassDeletedEvent struct {
-	ingressClass *slim_networkingv1.IngressClass
+	wasDefault bool
 }
 
-func newIngressClassManager(clientset k8sClient.Clientset, ingressQueue workqueue.RateLimitingInterface, maxRetries int) (*ingressClassManager, error) {
+type ingressClassManager struct {
+	isDefaultIngressClass atomic.Bool
+	synced                atomic.Bool
+	ingressClasses        resource.Resource[*slim_networkingv1.IngressClass]
```

Comment:
looks like the field `ingressClasses` isn't used and can be removed

Reply:
Good catch!